src/nvim/context.h

```diff
@@ -24,21 +25,54 @@ typedef kvec_t(Context) ContextVec;
   .regs = MSGPACK_SBUFFER_INIT, \
   .jumps = MSGPACK_SBUFFER_INIT, \
   .buflist = MSGPACK_SBUFFER_INIT, \
-  .gvars = MSGPACK_SBUFFER_INIT, \
+  .vars = MSGPACK_SBUFFER_INIT, \
   .funcs = ARRAY_DICT_INIT, \
 }
 
 typedef enum {
-  kCtxRegs = 1,       ///< Registers
-  kCtxJumps = 2,      ///< Jumplist
-  kCtxBuflist = 4,    ///< Buffer list
-  kCtxGVars = 8,      ///< Global variables
-  kCtxSFuncs = 16,    ///< Script functions
-  kCtxFuncs = 32,     ///< Functions
+  kCtxRegs    =   (1 <<  0),  ///< Registers
+  kCtxJumps   =   (1 <<  1),  ///< Jumplist
+  kCtxBuflist =   (1 <<  2),  ///< Buffer list
+  kCtxSVars   =   (1 <<  3),  ///< Script-local variables
+  kCtxGVars   =   (1 <<  4),  ///< Global variables
+  kCtxBVars   =   (1 <<  5),  ///< Buffer variables
+  kCtxWVars   =   (1 <<  6),  ///< Window variables
+  kCtxTVars   =   (1 <<  7),  ///< Tab variables
+  kCtxLVars   =   (1 <<  8),  ///< Function-local variables
+  kCtxSFuncs  =   (1 <<  9),  ///< Script functions
+  kCtxFuncs   =   (1 << 10),  ///< All functions
 } ContextTypeFlags;
 
+extern int kCtxVars;
 extern int kCtxAll;
 
+#define CONTEXT_TYPE_FROM_STR(types, str) \
```

Comment:
seems like this is used in only 1 place. better to avoid macros where possible.

Reply:
Two places, `eval.c` and `vim.c`, in general anywhere I want to make a flags variable from some context type strings